### ISSUE TITLE
RX based sort throws error.

### ISSUE THREAD

**Issue by ButchersBoy** [RolandPheasant/DynamicData#64]
```SortController ``` is now marked obsolete.  When I try and use the preferred RX based overloads of ``` .Sort ```, the method throws an ``` ArgumentNullException ```.

You can reproduce this easily by by creating a new Console App, and with the following **Program.cs** contents:

```
using System;
using System.Collections.ObjectModel;
using System.Reactive;
using System.Reactive.Subjects;
using DynamicData;
using DynamicData.Binding;

namespace DynamicDataSortWithRx
{
    public class Data
    {
        public Data(int id, string value)
        {
            Id = id;
            Value = value;
        }

        public int Id { get; }
        public string Value { get; }
    }

    class Program
    {
        static void Main(string[] args)
        {
            var cache = new SourceCache<Data, int>(d => d.Id);

            var sortPump = new Subject<Unit>();

            ReadOnlyObservableCollection<Data> datas;
            var disposable = cache.Connect()
                .Sort(SortExpressionComparer<Data>.Ascending(d => d.Id), sortPump)
                .Bind(out datas)
                .Subscribe();

            disposable.Dispose();
        }
    }
}
```

Stack trace:

```
   at System.Reactive.Linq.Observable.Synchronize[TSource](IObservable`1 source, Object gate)
   at DynamicData.Cache.Internal.Sort`2.<Run>b__7_0(IObserver`1 observer)
   at System.Reactive.AnonymousObservable`1.SubscribeCore(IObserver`1 observer)
   at System.Reactive.ObservableBase`1.Subscribe(IObserver`1 observer)
--- End of stack trace from previous location where exception was thrown ---
   at System.Reactive.PlatformServices.ExceptionServicesImpl.Rethrow(Exception exception)
   at System.Reactive.Stubs.<>c.<.cctor>b__2_1(Exception ex)
   at System.Reactive.AnonymousObserver`1.OnErrorCore(Exception error)
   at System.Reactive.ObserverBase`1.OnError(Exception error)
   at System.Reactive.SafeObserver`1.OnError(Exception error)
   at System.Reactive.Linq.ObservableImpl.Do`1._.OnError(Exception error)
   at System.Reactive.AutoDetachObserver`1.OnErrorCore(Exception exception)
   at System.Reactive.ObserverBase`1.Fail(Exception error)
   at System.Reactive.ObservableBase`1.Subscribe(IObserver`1 observer)
   at System.ObservableExtensions.SubscribeSafe[T](IObservable`1 source, IObserver`1 observer)
   at System.Reactive.Linq.ObservableImpl.Do`1.Run(IObserver`1 observer, IDisposable cancel, Action`1 setSink)
   at System.Reactive.Producer`1.Run(IScheduler _, State x)
   at System.Reactive.Concurrency.ScheduledItem`2.InvokeCore()
   at System.Reactive.Concurrency.CurrentThreadScheduler.Trampoline.Run(SchedulerQueue`1 queue)
   at System.Reactive.Concurrency.CurrentThreadScheduler.Schedule[TState](TState state, TimeSpan dueTime, Func`3 action)
   at System.Reactive.Concurrency.LocalScheduler.Schedule[TState](TState state, Func`3 action)
   at System.Reactive.Producer`1.SubscribeRaw(IObserver`1 observer, Boolean enableSafeguard)
   at System.Reactive.Producer`1.Subscribe(IObserver`1 observer)
   at System.ObservableExtensions.Subscribe[T](IObservable`1 source)
   at DynamicDataSortWithRx.Program.Main(String[] args) in H:\GitHub\DynamicDataSortWithRx\DynamicDataSortWithRx\Program.cs:line 35
   at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
   at System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
   at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
```

**Comment by RolandPheasant** [comment]
I thought this was fixed.  Will check it out

**Comment by RolandPheasant** [comment]
I can see what it is. Should be fixed in the next few days

**Comment by RolandPheasant** [comment]
@ButchersBoy fix deployed to  4.10.1 and 5.1.4
